libs/content/workspace/src/space_inspector/data.rs

```diff
@@ -0,0 +1,160 @@
+use lb_rs::model::api::FileUsage;
+use lb_rs::model::file::File;
+use lb_rs::Uuid;
+use serde::Deserialize;
+use std::collections::HashMap;
+
+/// Contains data related to folders and files needed for space inspector
+#[derive(Debug, Clone, Default, PartialEq)]
+pub struct Data {
+    pub focused_folder: Uuid,
+    pub all_files: HashMap<Uuid, FileRow>,
+    pub folder_sizes: HashMap<Uuid, u64>,
+    pub root: Uuid,
+}
+
+#[derive(Debug, Deserialize, Clone, Hash, PartialEq, Eq)]
+pub struct FileRow {
+    pub file: File,
+    pub size: u64,
+}
+
+/// Stores information in a tree format returned by get_children()
+#[derive(PartialEq, Debug, Clone)]
+pub struct StorageTree {
+    id: Uuid,
+    name: String,
+    portion: f32,
+    children: Vec<StorageTree>,
+}
+
+/// Responsible for storing relevant folder information for painting
+#[derive(PartialEq, Debug, Clone)]
+pub struct StorageCell {
+    pub id: Uuid,
+    pub name: String,
+    pub portion: f32,
+    pub layer: u64,
+}
+
+impl Data {
+    pub fn init(potential_root: Option<File>, usage: Vec<FileUsage>, meta_data: Vec<File>) -> Self {
+        let data = Self::get_filerows(usage, meta_data);
+        let mut all_files = HashMap::new();
+        let mut root = Uuid::nil();
+        for datum in data.clone() {
+            if datum.file.is_root() {
+                root = datum.file.id;
+            }
+            all_files.insert(datum.file.id, datum);
+        }
+
```

Comment:
imo if root == nil at this point you should `panic!`. Otherwise you will have an infinite loop slightly below.